### PR TITLE
treewide: Add `meta.pkgConfigModules` to a few packages

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Description of changes

Picking up where #215056 left off.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/gnome/core/libgnome-keyring/default.nix [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
   pname = "libgnome-keyring";
   version = "3.12.0";
 in
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   name = "${pname}-${version}";
 
   src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${name}.tar.xz";
+    url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
```
Comment (roberth) [inline pkgs/desktops/gnome/core/libgnome-keyring/default.nix:13]: Missing `let inherit`?

**Comment by roberth** [comment]
I think the tester could be improved by checking that the `meta` attribute is complete.
I think I've spotted a small mistake, but otherwise lgtm.

File: pkgs/desktops/gnome/core/libgnome-keyring/default.nix (hunk 1 above)
Comment (Ericson2314) [reply pkgs/desktops/gnome/core/libgnome-keyring/default.nix:13]: That one is actually not an error, see the let binding above -- it is sort of half converted to use `pname` and `version`. It should be fully converted but in a second PR I think so this one stays no rebuilds.

**Comment by Ericson2314** [comment]
I think I will want to make the test more restrictive only after the current crop of pkgs is fixed :). That way I can at least briefly turn on the warnings before getting any new one.

(Though I suppose this would be a build-time failure.)

File: pkgs/desktops/gnome/core/libgnome-keyring/default.nix (hunk 1 above)
Comment (roberth) [reply pkgs/desktops/gnome/core/libgnome-keyring/default.nix:13]: I must have read it too quickly then.

**Comment by roberth** [comment]
Sounds good.

If everything is well documented and the error message explains itself, we might as well "risk it" at some point and let the one or two contributors who have a failing package deal with the little bit of extra work, especially if all the package are still in the hydra-tested set that's `defaultPkgConfigPackages`.

File: pkgs/development/libraries/qt-5/modules/qtbase.nix [hunk 2]
```diff
@@ -338,12 +339,33 @@ stdenv.mkDerivation {
...
+      "Qt5Gui"
+      "Qt5Network"
+      "Qt5OpenGL"
+      "Qt5OpenGLExtensions"
+      "Qt5PrintSupport"
+      #"Qt5Qml"
+      #"Qt5QmlModels"
+      #"Qt5Quick"
+      #"Qt5QuickTest"
+      #"Qt5QuickWidgets"
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qt-5/modules/qtbase.nix:362]: From the code it is not clear why those are commented

Comment (roberth) [reply pkgs/development/libraries/qt-5/modules/qtbase.nix:362]: We'll inevitably come back to this when changing the test to check for the completeness of the meta field. I'm not too worried about it, but next time an explanatory comment would be nice.

Comment (Ericson2314) [reply pkgs/development/libraries/qt-5/modules/qtbase.nix:362]: Yeah sorry there is not a comment. The issue is that these are listed in the JSON file but don't seem to be inside the package.

**Pull request merged**